cpp/src/arrow/compute/kernels/scalar_temporal.cc

```diff
@@ -451,9 +517,14 @@ const FunctionDoc day_doc{
 
 const FunctionDoc day_of_week_doc{
     "Extract day of the week number",
-    ("Week starts on Monday denoted by 0 and ends on Sunday denoted by 6.\n"
-     "Returns an error if timestamp has a defined timezone. Null values return null."),
-    {"values"}};
+    ("By default Week starts on Monday represented by 0 and ends on Sunday represented "
+     "by 6.\n"
+     "Returns an error if timestamp has a defined timezone. Null values return null.\n"
+     "DayOfWeekOptions.week_start can be used to set another starting day using ISO "
```

Comment:
Can you move this up to put it directly after the "By default, the week starts with ..", so before the "Returns an error if ...". Since this explanation of the options directly adds to the default monday-sunday as 0-6, putting the other sentence in between breaks a bit the logical flow I think

Reply:
Done. Please check if it looks right https://github.com/apache/arrow/pull/10598/commits/db63b17be020bd686f2c4d4494fe04b7e3175d99#diff-53850537a2f3f7977473b30241fdd3ddb35aa97269284c06471d9b0c599e2c59.